[UPLOAD-CRASH] 在服务端加载客户端 mod

### 描述问题

```bash
如题
附上chatgpt的分析，导致错误的mod分析的是正确的，细节就不清楚了


根据日志显示，服务器崩溃的主要原因是与Mod加载相关的问题，具体而言，是由于一个名为"Distant Horizons"的Mod加载失败导致的。在日志的末尾有一段错误消息:


net.minecraftforge.fml.LoadingFailedException: Loading errors encountered: [
    Distant Horizons (distanthorizons) has failed to load correctly
    java.lang.reflect.InvocationTargetException: null
]
```

这表明"Distant Horizons" Mod加载时出现了问题。进一步的错误信息表明，在创建该Mod实例时发生了反射调用异常。

```
Caused by: java.lang.ExceptionInInitializerError
    at loaderCommon.forge.com.seibel.distanthorizons.common.wrappers.DependencySetup.createClientBindings(DependencySetup.java:69) ~[DistantHorizons-2.0.1-a-1.18.2.jar%2363!/:?]
    at com.seibel.distanthorizons.forge.ForgeMain.<init>(ForgeMain.java:98) ~[DistantHorizons-2.0.1-a-1.18.2.jar%2363!/:?]
    ... 11 more
Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/Minecraft for invalid dist DEDICATED_SERVER
```

这表明该Mod试图在服务器端加载客户端特定的类(net.minecraft.client.Minecraft)，这是不合理的，因为服务器端不应该加载客户端的类。

要解决这个问题，您可以尝试联系Mod作者或社区以获取支持，他们可能已经意识到了这个问题并提供了解决方案或更新。另外，您还可以尝试禁用或删除"Distant Horizons" Mod来查看服务器是否可以正常启动。
```


### 您正在进行的操作是？

```bash
1. 
2. 
3. 
4. 游戏崩溃
```


### 日志文件

https://mclo.gs/FcvT9q0

### 操作系统

Windows

### 客户端 / 服务端

服务端

--- a/docs/.vitepress/analyzers/mcla.ts
+++ b/docs/.vitepress/analyzers/mcla.ts
@@ -12,7 +12,7 @@ import type {
 } from "./mcla.api"
 export * from "./mcla.api"
 
-const VERSION = "v0.4.23"
+const VERSION = "v0.5.0"
 // const VERSION = "dev"
 const RESOURCES_BASE = "https://globemc.github.io/mcla"
 const GO_WASM_EXEC_URL = useCDN(`${RESOURCES_BASE}/${VERSION}/wasm_exec.js`)
@@ -226,7 +226,7 @@ async function loadMCLA(loadProgress?: Ref<number>): Promise<MCLAAPI> {
 
   console.log("Loading MCLA ...")
 
-  await import(GO_WASM_EXEC_URL /* @vite-ignore */) // set variable `window.Go`
+  await import(/* @vite-ignore */ GO_WASM_EXEC_URL) // set variable `window.Go`
 
   const go = new (window as any as containsGoCls).Go()
   var res

--- a/docs/.vitepress/theme/components/Analyzer.vue
+++ b/docs/.vitepress/theme/components/Analyzer.vue
@@ -1147,6 +1147,9 @@ onUnmounted(() => {
                   匹配度: {{ match.match * 100 }}%
                 </span>
               </h5>
+              <code class="result-parsed-error">
+                {{ match.desc.message }}
+              </code>
               <div v-for="(sol, n) in match.desc.solutions" :key="n">
                 <div v-if="sol.ok">
                   <details :open="false">
@@ -1159,7 +1162,7 @@ onUnmounted(() => {
                     </div>
                     <div>
                       <b>解决方案: </b>
-                      <a @click.prevent="redirectTo(sol.res.link_to, true)">
+                      <a target="_blank" :href="sol.res.link_to">
                         打开文档 <OpenTabIcon />
                       </a>
                     </div>

--- a/docs/.vitepress/workers/mcla.worker.ts
+++ b/docs/.vitepress/workers/mcla.worker.ts
@@ -188,7 +188,7 @@ async function setup(data: SetupData) {
   )
 
   console.debug(`Loading MCLA from ${data.wasm_url} ...`)
-  await import(data.wasm_exec /* @vite-ignore */) // set variable `globalThis.Go`
+  await import(/* @vite-ignore */ data.wasm_exec) // set variable `globalThis.Go`
   const go = new globalThis.Go()
   var res
   if (WebAssembly.instantiateStreaming) {

--- a/docs/server/mods.md
+++ b/docs/server/mods.md
@@ -1,5 +1,21 @@
 # Mod 问题
 
+## 在服务端运行仅客户端模组
+
+解决方案：移除日志中指示的 Mod。例如下方的情况应该移除 distanthorizons。
+
+```
+[net.minecraftforge.fml.javafmlmod.FMLModContainer/LOADING]: Failed to create mod instance. ModID: distanthorizons, class com.seibel.distanthorizons.forge.ForgeMain
+```
+
+关键词：
+
+```
+Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/Minecraft for invalid dist DEDICATED_SERVER
+
+java.lang.NoClassDefFoundError: net/minecraft/client/Minecraft
+```
+
 ## Fabric
 
 ### 模组问题

